src/context/WalletProvider/KeepKey/components/Connect.tsx

```diff
@@ -55,80 +55,66 @@ export const KeepKeyConnect = () => {
     setError(null)
     setLoading(true)
 
-    const firstAdapter = (await getAdapter(KeyManager.KeepKey)) as KkRestAdapter | null
-    if (firstAdapter) {
-      const wallet = await (async () => {
-        try {
-          const sdk = await setupKeepKeySDK()
-          if (!sdk) throw new Error('Failed to setup KeepKey SDK')
-          const wallet = await firstAdapter.pairDevice(sdk)
-          if (!wallet) {
-            setErrorLoading('walletProvider.errors.walletNotFound')
-            return
-          }
-          return wallet
-        } catch (e) {
+    const wallet = await (async () => {
+      try {
+        const sdk = await setupKeepKeySDK()
+
+        // There is no need to instantiate KkRestAdapter and attempt pairing if SDK is undefined
+        // It will only be defined in the context of KK desktop app, and is irrelevant otherwise
+        if (sdk) {
+          const firstAdapter = (await getAdapter(KeyManager.KeepKey)) as KkRestAdapter | null
+          return await firstAdapter?.pairDevice(sdk)
+        } else {
           const secondAdapter = await getAdapter(KeyManager.KeepKey, 1)
           // @ts-ignore TODO(gomes): FIXME, most likely borked because of WebUSBKeepKeyAdapter
-          const wallet = await secondAdapter.pairDevice().catch(err => {
-            if (err.name === 'ConflictingApp') {
-              setErrorLoading('walletProvider.keepKey.connect.conflictingApp')
-              return
-            }
-            console.error(e)
-            setErrorLoading('walletProvider.errors.walletNotFound')
-            return
-          })
-          if (!wallet) {
-            setErrorLoading('walletProvider.errors.walletNotFound')
-            return
-          }
-          return wallet
+          return await secondAdapter?.pairDevice()
```

Comment:
The guts of this, IIAFE `wallet` which is *not* attempting to get the first adapter in case there's no SDK. The diff below is just moved pieces